api/src/org/apache/cloudstack/api/command/admin/vpc/VPCOSPFConfigUpdateCmd.java

```diff
@@ -0,0 +1,207 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.api.command.admin.vpc;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.cloudstack.api.APICommand;
+import org.apache.cloudstack.api.ApiConstants;
+import org.apache.cloudstack.api.ApiErrorCode;
+import org.apache.cloudstack.api.BaseCmd;
+import org.apache.cloudstack.api.Parameter;
+import org.apache.cloudstack.api.ServerApiException;
+import org.apache.cloudstack.api.response.VPCOSPFConfigResponse;
+import org.apache.cloudstack.api.response.ZoneResponse;
+import org.apache.log4j.Logger;
+
+import com.cloud.network.vpc.OSPFZoneConfig;
+import com.cloud.network.vpc.VpcProvisioningService;
+import com.cloud.user.Account;
+import com.cloud.utils.net.cidr.BadCIDRException;
+
+@APICommand(name = "vpcOSPFConfigUpdate", description = "Captures config informaton for ospf zone level params", responseObject = VPCOSPFConfigResponse.class, since = "4.9.0", requestHasSensitiveInfo = true)
+public class VPCOSPFConfigUpdateCmd extends BaseCmd {
+    public static final Logger s_logger = Logger.getLogger(VPCOSPFConfigUpdateCmd.class);
+    private static final String s_name = "vpcospfconfigresponse";
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+
+    @Parameter(name = ApiConstants.ZONE_ID, type = CommandType.UUID, entityType = ZoneResponse.class, required = true, description = "the ID of the Zone")
+    private Long zoneid;
+
+    @Parameter(name = OSPFZoneConfig.s_protocol, type = CommandType.STRING, required = false, description = "the password used to secure inter ospf communication(default ospf)")
+    private String protocol;
+
+    @Parameter(name = OSPFZoneConfig.s_area, type = CommandType.STRING, required = false, description = "Specify the OSPF Area ID")
+    private String area;
+
+    @Parameter(name = OSPFZoneConfig.s_helloInterval, type = CommandType.SHORT, required = false, description = "Set number of seconds for HelloInterval timer value.(default 10)")
```

Comment:
There is no default value in Parameter annotation. How will that work, like if in request a field is empty fill it with default value if given ? I think it may not be very useful.